File: solutions/2026-08-03/solution.java
// Longest Substring Without Repeating Characters
// Difficulty: Medium
// URL: https://leetcode.com/problems/longest-substring-without-repeating-characters/

class Solution {
  public int lengthOfLongestSubstring(String s) {
    Set<Character> charSet = new HashSet<>();
    int left = 0;
    int res = 0;
    for (int right = 0; right < s.length(); right++) {
      while (charSet.contains(s.charAt(right))) {
        charSet.remove(s.charAt(left));
        left++;
      }
      charSet.add(s.charAt(right));
      res = Math.max(res, right - left + 1);
    }
    return res;
  }
}

File: solutions/2026-08-03/solution.py
# Longest Substring Without Repeating Characters
# Difficulty: Medium
# URL: https://leetcode.com/problems/longest-substring-without-repeating-characters/

def lengthOfLongestSubstring(s: str) -> int:
  char_set = set()
  left = 0
  res = 0
  for right in range(len(s)):
    while s[right] in char_set:
      char_set.remove(s[left])
      left += 1
    char_set.add(s[right])
    res = max(res, right - left + 1)
  return res